src/main/java/com/rackspace/feeds/feedscatalog/ResolveHostFilter.java

```diff
@@ -0,0 +1,138 @@
+     */
+    public void init(FilterConfig filterConfig) throws ServletException {
+        LOG.debug( "initializing ResolveHostFilter" );
+
+        try {
+            //load the resolve host xslt from resource
+            URL xsltResource = Thread.currentThread().getContextClassLoader().getResource(resolveHostXslt);
+            if (xsltResource != null) {
+                transformer = TransformerUtils.getInstanceForXsltAsFile(xsltResource.getPath());
+            }
+            else {
+                throw new UnavailableException("resource is null - " + resolveHostXslt);
+            }
+        } catch ( Exception e ) {
+            LOG.error( "Error loading xslt: " + e.getMessage());
+            throw new ServletException( e );
+        }
+    }
+
+    public void doFilter (ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+
+        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+        HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+
+        // setup wrapper response with output stream to collect transformed content
+        ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        ServletOutputStreamWrapper outputStreamWrapper = new ServletOutputStreamWrapper(stream);
+        OutputStreamResponseWrapper wrappedResponse =
+                new OutputStreamResponseWrapper(httpServletResponse, outputStreamWrapper);
+
+        // apply filter further down the chain on wrapped response
+        chain.doFilter(httpServletRequest, wrappedResponse);
+
+        // obtain response content
+        String originalResponseContent = stream.toString();
+        LOG.debug("Original response content length = " + originalResponseContent.length());
+
+        if (StringUtils.isNotEmpty(originalResponseContent)) {
+
+            // create input params to xslt with headers from request
+            HashMap<String, Object> params = new HashMap<String, Object>();
+            putTenantIdParams(httpServletRequest, params);
+            putExternalLocParams(httpServletRequest, params);
+
+            try {
+                OutputStream outputStream = new ByteArrayOutputStream();
+
+                // transform response content with resolve host xslt
+                transformer.doTransform(params,
+                    new StreamSource(new StringReader(originalResponseContent)),
+                    new StreamResult(outputStream));
+
+                // set transformed content to response
+                String newResponseContent = outputStream.toString();
+                LOG.debug("New response content length = " + newResponseContent.length());
+                httpServletResponse.setContentLength(newResponseContent.length());
+                httpServletResponse.getWriter().write(newResponseContent);
+            }
+            catch (Exception e) {
+                LOG.error("Error transforming xml: " + e.getMessage());
```

Comment:
If there's an error transforming the xml, we should throw the exception. I.e: just pass the exception to the caller (the servlet engine) so it knows to return 500 error to clients.